### PR TITLE
Fix minor issues in dev portal and tenant mode

### PR DESCRIPTION
**Pull request by inthirakumaaran**
## Purpose
- Add button and tooltip to the URL component.
image

- Separate out general setting component and form

- Get super tenant name from the backend.

- Add APIs call client id revocation and regeneration

### CODE REVIEW

File: apps/admin-portal/src/components/applications/components/url-input-component.tsx [hunk 1]
```diff
@@ -57,10 +58,11 @@ export const URLInputComponent: FunctionComponent<URLInputComponentInterface> =
...
     const [duplicateURL, setDuplicateURL] = useState(false);
+    const [keepFocus, setKeepFocus] = useState(false);
 
     /**
      * Enter button option.
-     * @param e Enter event.
+     * @param e keypress event.
      */
     const keyPressed = (e) => {
         if (e.keyCode === 13) {
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/components/url-input-component.tsx]:
```suggestion
        const key = e.which || e.charCode || e.keyCode;
        if (key === 13) {
```

File: apps/admin-portal/src/components/applications/components/url-input-component.tsx [hunk 2]
```diff
@@ -120,6 +142,11 @@ export const URLInputComponent: FunctionComponent<URLInputComponentInterface> =
         }
     };
 
+    const addFromButton = (e) => {
+        e.preventDefault();
+        addUrl()
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/components/url-input-component.tsx]:
```suggestion
        addUrl();
```

Missing semi colon

File: apps/admin-portal/src/components/applications/components/url-input-component.tsx [hunk 3]
```diff
@@ -158,29 +185,33 @@ export const URLInputComponent: FunctionComponent<URLInputComponentInterface> =
...
+                        error={ validURL && !duplicateURL ? false : true }
+                        focus={ keepFocus }
+                        value={ changeUrl }
+                        onKeyDown={ keyPressed }
+                        onChange={ handleChange }
+                        onBlur={ handleOnBlur }
+                        placeholder={ placeholder }
+                        action
+                    >
+                        <input/>
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/components/url-input-component.tsx]: Why do we need this empty input?

File: apps/admin-portal/src/components/applications/forms/general-details-form.tsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        name,
+        description,
+        discoverability,
+        imageUrl,
+        accessUrl,
+        onSubmit
+    } = props;
+
+    const [isDiscoverable, setDiscoverability] = useState<boolean>(discoverability);
+
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/forms/general-details-form.tsx]:
```suggestion
```

File: apps/admin-portal/src/components/applications/components/url-input-component.tsx [hunk 5]
```diff
@@ -158,29 +185,33 @@ export const URLInputComponent: FunctionComponent<URLInputComponentInterface> =
...
+                                (
+                                    <Button
+                                        onClick={ (e) => addFromButton(e) }
+                                        icon="add"
+                                        type="button"
+                                    />
+                                )
+                            }
+                            position="top center"
+                            content="Add URL."
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/components/url-input-component.tsx]:
```suggestion
                            content="Add URL"
```

File: apps/admin-portal/src/components/applications/components/url-input-component.tsx (hunk 3 above)
Comment (inthirakumaaran) [reply apps/admin-portal/src/components/applications/components/url-input-component.tsx]: When we write multiple actions then in the children we need to define this to get the input box. This is how defined in semantic https://react.semantic-ui.com/elements/input/#variations-actions

Comment (jeradrutnam) [reply apps/admin-portal/src/components/applications/components/url-input-component.tsx]: ok

**Pull request merged**